### PR TITLE
ENH: Add box plot

### PR DESCRIPTION
**Pull request by pratapvardhan**
```python
pd.DataFrame(np.random.randn(100, 5), columns=['A', 'B', 'C', 'D', 'E']).plot.box()
```
image

### CODE REVIEW

File: altair_pandas/_core.py [hunk 1]
```diff
@@ -62,6 +62,15 @@ def hist(self, **kwargs):
             y=alt.Y('count()', title='Frequency'),
         )
 
+    def box(self, **kwargs):
+        data = self._data.iloc[:, 1:]
+        return alt.Chart(data).transform_fold(
+            list(data.columns), as_=['column', 'value']
+        ).mark_boxplot().encode(
+            x=alt.X('column:N', title=''),
```
Comment (jakevdp) [inline altair_pandas/_core.py]: I'd use ``title=None`` rather than ``title=''``.

File: altair_pandas/_core.py [hunk 2]
```diff
@@ -137,6 +146,15 @@ def hist(self, **kwargs):
             color=alt.Color('column:N')
         )
 
+    def box(self, **kwargs):
+        data = self._data.iloc[:, 1:]
+        return alt.Chart(data).transform_fold(
+            list(data.columns), as_=['column', 'value']
+        ).mark_boxplot().encode(
+            x=alt.X('column:N', title=''),
```
Comment (jakevdp) [inline altair_pandas/_core.py]: Same here.

File: altair_pandas/test_plotting.py [hunk 3]
```diff
@@ -63,3 +63,14 @@ def test_dataframe_hist(dataframe, with_plotting_backend):
...
+
+def test_boxplot(series, dataframe, with_plotting_backend):
+    for data in [series, dataframe]:
+        chart = data.plot.box()
+        spec = chart.to_dict()
+        assert spec['mark'] == 'boxplot'
+        assert spec['encoding']['x']['field'] == 'column'
+        assert spec['encoding']['y']['field'] == 'value'
+        fold = ['data_name'] if isinstance(data, pd.Series) else ['x', 'y']
+        assert spec['transform'][0]['fold'] == fold
```
Comment (jakevdp) [inline altair_pandas/test_plotting.py]: Can you split this into two functions? The reason is, if one starts to fail it's more clear exactly where the failure is. Here if series.plot.box() starts to fail, you have to add logging and re-run to figure out whether it was the series plot or the dataframe plot (or both) that failed.

**Comment by pratapvardhan** [comment]
@jakevdp -- understood, review comments are resolved now.

**Comment by jakevdp** [comment]
Thanks!

**Pull request merged**